sdk/python/lib/pulumi/resource.py

```diff
@@ -683,7 +691,7 @@ def __init__(self,
         # Collapse any `Alias`es down to URNs. We have to wait until this point to do so because we
         # do not know the default `name` and `type` to apply until we are inside the resource
         # constructor.
-        self._aliases = []
+        self._aliases: List[Input[str]] = [] # pylint: disable=used-before-assignment
```

Comment:
Why do we need a pylint disable here?

Reply:
This was related to the commend you made below (and what I discovered through the test failures) that the types should generally go inside of quotes. The Input import is inside of an `if TYPE_CHECKING:` statement